sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -215,22 +228,35 @@ public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse
      * It will call cancelOperation if status is {@link OperationStatus#IN_PROGRESS} otherwise it does nothing.
      *
      * @throws UnsupportedOperationException when the cancel operation is not supported by the Azure service.
+     * @return A {@link Mono} containing the response.
      */
-    public void cancelOperation() throws UnsupportedOperationException {
+    public Mono<T> cancelOperation() throws UnsupportedOperationException {
         if (this.cancelOperation == null) {
             throw logger.logExceptionAsError(new UnsupportedOperationException(
                 "Cancel operation is not supported on this service/resource."));
         }
 
         // We can not cancel an operation if it was never started
         // It only make sense to call cancel operation if current status IN_PROGRESS.
+
         final PollResponse<T> response = this.pollResponse;
         if (response != null && response.getStatus() != OperationStatus.IN_PROGRESS) {
-            return;
+            return Mono.empty();
         }
 
-        // Time to call cancel
-        this.cancelOperation.accept(this);
+        //Time to call cancel
+        return this.cancelOperation.apply(this);
```

Comment:
What is fcancelOepration throws an exception>

Reply:
If its a REST call in cancelOperation then the behavoiur will be same as an API call.
User can specify .onError on the Mono returned to them from cancelOperation.
if needed in CancelOperation, Mono.error should be returned there.